### ISSUE TITLE
A configuration file is required for --help

### ISSUE THREAD

**Issue by babolivier** [matrix-org/synapse#7389]
Synapse complains about missing a config file if called with `--help`, instead of just printing the help doc:

```
$ python -m synapse.app.homeserver --help
usage: homeserver.py [-c CONFIG_FILE] [--generate-config] [--generate-missing-configs] [-H SERVER_NAME] [--report-stats {yes,no}] [--config-directory DIRECTORY] [--data-directory DIRECTORY] [--open-private-ports]
homeserver.py: error: Must supply a config file.
A config file can be automatically generated using "--generate-config -H SERVER_NAME -c CONFIG-FILE"
```

If a configuration file is provided, it will print the help doc as expected:

```
$ python -m synapse.app.homeserver -c homeserver.yaml --help
usage: homeserver.py [-h] [-c CONFIG_FILE] [--generate-config] [--generate-missing-configs] [-H SERVER_NAME] [--report-stats {yes,no}] [--config-directory DIRECTORY] [--data-directory DIRECTORY] [--open-private-ports] [-D]
                     [--print-pidfile] [--manhole PORT] [-d SQLITE_DATABASE_PATH] [-n] [--enable-registration]

Synapse Homeserver

optional arguments:
  -h, --help            show this help message and exit
  -c CONFIG_FILE, --config-path CONFIG_FILE
                        Specify config file. Can be given multiple times and may specify directories containing *.yaml files.
[...]
```

It seems a bit silly to me that we need a configuration file just to display the help for `synapse.app.homeserver`.

**Comment by richvdh** [comment]
I believe this got fixed by #7249

**Comment by babolivier** [comment]
Wow I really expected my memory's lifespan to be better than that.
